### DISCUSSION TITLE
Grove - GSR Sensor | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#241] [Q&A]
# Grove - GSR Sensor | Seeed Studio Wiki

Grove - GSR Sensor

https://wiki.seeedstudio.com/Grove-GSR_Sensor/

**Comment by Dazzshiet** [comment]
Can you help me I try to maximize my output to 600 by setting the resistor but instead go lower to 5 when I am trying the opposite way it's doesn't really effect the output, can you with that please answer this is urgent

**Reply by MatthewJeffson** [reply]
It looks like is broken, have you tried to contact the [tech support](https://www.seeedstudio.com/contacts)?

**Comment by Dazzshiet** [comment]
Can anyone help me if I place my finger and give some water or lotion to make it wet , the output value go down instead go up is it supposed to work like that or no?

**Reply by xzxcessarr** [reply]
Hello, from your description we preliminary think it is well: because the Human Resistance of the wet skin is lower than the dry. While, if you wanna test if your module is well or not, we recommand you put the sense to your fingertips to analysis if the Human Resistance of it is higher than other part of your hand.

**Comment by devdattatemgire** [comment]
Do i need to have Grove Base Hat for RasPi or can the sensor be connected to RasPi directly just like it can be done in Arduino

**Reply by xzxcessarr** [reply]
If well, we recommand you to use the shield like this one: https://wiki.seeedstudio.com/Grove_Base_Hat_for_Raspberry_Pi/
but if you just wanna test it, you can use the dupont line or other jumper line to connect

**Comment by mahsamirheydarian** [comment]
Can we connect the sensor to the laptop without the base shield?

**Comment by mahsamirheydarian** [comment]
I need the scl index to do architectural research, how can I get this index?

**Comment by devdattatemgire** [comment]
I recently purchased a Grove GSR sensor and Base Hat for Raspberrypi  for my academic project, where I aim to measure and analyze the stress levels of individuals. I have a couple of queries regarding the sensor and its readings that I hope you can help me clarify.

Unit of Readings:
I would like to confirm the unit of the readings obtained from the GSR sensor. Are the readings in Volts (V) or Millivolts (mV)? This information is crucial for the accurate interpretation of the data in my project.

Formula for Resistance:
The documentation provided with the sensor includes a formula for calculating resistance based on the readings from the sensor. I wanted to inquire about the accuracy of this formula. Does the formula reliably calculate the resistance, and if so, what is the unit of the obtained resistance readings? Is it in Ohms (Ω) or Milliohms (mΩ)?

Additionally, I have followed the calibration process outlined in the documentation and set the initial readings to 512. If there are any additional tips or considerations for calibration specific to this sensor, I would greatly appreciate any guidance.

Your assistance in providing clarity on these matters will significantly contribute to the success of my academic project.

**Reply by Alfred-lms** [reply]
Hi, although I'm not a developer of the sensor in question, i can speak a little bit about the arduino side of things. The arduino reads the value of the sensor through a Analogic port, which means it reads the voltage the IC on the shield sends as signal, i'll be doing some tests with a oscilloscope but as a upfront answer, the ADC of the arduino is a 10bit one, which means you can get a reading from 0-5v/1024 ~= 0.005V per step. If you're able to isolate the data to read perfectly this stepping this should at least help you with your interpretation.

**Comment by KeithDewey** [comment]
This is cool!
Plugging everything in was surprisingly simple!
Two questions that you may be able to help me with...
1) Does it matter which fingers the sensors go on? Same hand?
2) Above, you say "adjust resistor until the serial output is minimized". I can see the screw. How do I know if the serious output is minimized?
Many thanks (the project will be used to measure stress for therapy clients and for an MSc Health Psychology!)

**Reply by KeithDewey** [reply]
More details about my GSR psychological stress testing project can be found here: https://keithdewey.com/articles/innovation/galvanic-skin-response-gsr-sensor-homemade-stress-monitor/

**Comment by volkere** [comment]
error message when running grove_gsr_sensor.py code: from grove.adc import ADC not found. 
Appreciate any advice

**Reply by grilliiiii** [reply]
Are you using the Grove Base Hat for Raspberry Pi? If so, please confirm whether the chip on your board is MM32 or STM32 (if you’re unsure, you can upload a picture of your chip).

If it's the STM32 version (V1.1), the I2C address is 0x04; you need to change the I2C address in /home/username/.local/lib/python3.9/site-packages/adc.py to 0x04.
If it's the MM32 version (V1.0), the I2C address is 0x08; you need to change the I2C address in /home/username/.local/lib/python3.9/site-packages/adc.py to 0x08.

The purpose of this is to ensure the I2C address is correctly matched. If you can't find this file path, please change the I2C address globally in all instances of adc.py
